src/simd/sq8_uniform_simd.cpp

```diff
@@ -0,0 +1,40 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "sq8_uniform_simd.h"
+
+#include "simd_status.h"
+
+namespace vsag {
+
+static SQ8UniformComputeCodesType
+SetSQ8UniformComputeCodesIP() {
+    if (SimdStatus::SupportAVX512()) {
+#if defined(ENABLE_AVX512)
```

Comment:
Can the values of `SimdStatus::SupportAVX512()` and `ENABLE_AVX512` be different sometimes?

Reply:
in compile time or at running